app/models/async_request/job.rb

```diff
@@ -1,6 +1,44 @@
 module AsyncRequest
-  class Job < ActiveRecord::Base
+  class Job < ActiveRecord::Base # rubocop:disable Rails/ApplicationRecord
     serialize :params, Array
-    enum status: [:waiting, :processing, :processed]
+    enum status: { waiting: 0, processing: 1, processed: 2, failed: 3 }
+
+    def self.execute_async(worker_class, *params)
+      raise ArgumentError if worker_class.nil?
+      job = create_and_enqueue(worker_class, *params)
+      [job, JsonWebToken.encode(job.id)]
+    end
+
+    def self.create_and_enqueue(worker_class, *params)
```

Comment:
if you are returning the job inside the previous method, you should not make this method part of the public API.
Apart from that, I don't really like the idea of returning an array, because you will always ignore the job. I don't know how we can handle this because this method was made public because I needed in some specs.